app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,368 @@
+
+  memoize
+  def concept_taught_exercise
+    Exercise::TaughtConcept.
+      where(exercise: active_concept_exercises).
+      pluck(:track_concept_id, :exercise_id).
+      map { |concept_id, exercise_id| { concept_id => exercise_id } }.
+      reduce({}, :merge)
+  end
+
+  def concept_exercises
+    {
+      num_exercises: active_concept_exercises.size,
+      num_exercises_target: concept_exercises_num_exercises_target,
+      created: active_concept_exercises.map { |exercise| serialize_exercise(exercise) }
+    }
+  end
+
+  def concept_exercises_num_exercises_target
+    NUM_CONCEPT_EXERCISES_TARGETS.find { |target| active_concept_exercises.size < target } || active_concept_exercises.size
+  end
+
+  def practice_exercises
+    {
+      num_exercises: active_practice_exercises.size,
+      num_exercises_target: practice_exercises_num_exercises_target,
+      created: active_practice_exercises.map { |exercise| serialize_exercise(exercise) },
+      volunteers: practice_exercises_volunteers,
+      health: practice_exercises_health
+    }
+  end
+
+  def practice_exercises_num_exercises_target
+    num_unimplemented = Track::UnimplementedPracticeExercises.(track.reload).size
+    max_target = track.num_exercises + num_unimplemented
+
+    NUM_PRACTICE_EXERCISES_TARGETS.find { |target| active_practice_exercises.size < target } || max_target
+  end
+
+  memoize
+  def num_unimplemented_practice_exercises
+    Track::UnimplementedPracticeExercises.(track).size
+  end
+
+  def practice_exercises_volunteers
+    authors = User.with_attached_avatar.where(id:
+      Exercise::Authorship.where(exercise: active_practice_exercises).select(:user_id))
+    contributors = User.with_attached_avatar.where(id:
+      Exercise::Contributorship.where(exercise: active_practice_exercises).select(:user_id))
+
+    serialize_volunteers(authors, contributors)
+  end
+
+  memoize
+  def practice_exercises_health
+    return :missing if active_practice_exercises.empty?
+    return :exemplar if active_practice_exercises.size >= NUM_PRACTICE_EXERCISES_TARGETS.last
+    return :needs_attention if active_practice_exercises.size < NUM_PRACTICE_EXERCISES_TARGETS.first
+
+    :healthy
+  end
+
+  memoize
+  def active_concept_exercises = track.concept_exercises.where(status: %i[active beta]).order(:position).to_a
+
+  memoize
+  def active_practice_exercises = track.practice_exercises.where(status: %i[active beta]).order(:position).to_a
+
+  def average_number_per_day(query, model)
+    total_count = query.where("#{model.table_name}.created_at >= ?", Time.current - NUM_DAYS_FOR_AVERAGE.days).count
+    (total_count / NUM_DAYS_FOR_AVERAGE.to_f).ceil
+  end
+
+  def percentage(count, total_count)
+    return 0.0 if total_count.zero?
+
+    ((count / total_count.to_f) * 100.0).round(1)
+  end
+
+  def average(count, total_count)
+    return 0 if total_count.zero?
+
+    (count.to_f / total_count).round(1)
```

Comment:
Experience tells me that nearly everything will be `1` or `2`. So I suspect rounding this to 1DP will be useful.

Reply:
Also in the UI can you say Mean or Median, not Average pls.